### PR TITLE
Formatting of all JS and Html files

### PR DESCRIPTION
**Pull request by aounhaider1**
Formatted all JS and HTML files of PaperUI.

Signed-off-by: Aoun Bukhari <bukhari@itemis.de>

### CODE REVIEW

File: extensions/ui/org.eclipse.smarthome.ui.paper/about.html [hunk 1]
```diff
@@ -2,84 +2,86 @@
     "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
```
Comment (kaikreuzer) [inline extensions/ui/org.eclipse.smarthome.ui.paper/about.html]: can we somehow avoid that the about.html is touched?

Comment (aounhaider1) [inline extensions/ui/org.eclipse.smarthome.ui.paper/about.html]: Yes, it can be excluded.

**Comment by GradyD** [comment]
Hey,

How did you format all of these? They appear to be using tabs or 8 spaces instead of 4 spaces. It also appears that the HTML is formatted in more of an XML style by breaking on the attributes instead of the element. 

Is there a reason that we feel the files need to be formatted?

**Comment by aounhaider1** [comment]
I used the ESH formatting as discussed here https://github.com/eclipse/smarthome/pull/987#issuecomment-181083298. PaperUI was recently converted to a JS project (https://github.com/eclipse/smarthome/pull/1009) to enable default formatting option on every 'Save'.

**Comment by kaikreuzer** [comment]
@GradyD See discussion here: https://github.com/eclipse/smarthome/pull/987#issuecomment-181026028

**Comment by GradyD** [comment]
I cloned your branch and the formatting is all off when I open the files in atom and also in eclipse. The HTML files break at odd spots in the middle of elements and do not always nest elements correctly, 
image

Not all of the indents but some are 1 tab and others are 2 tab indents. 

The javascript files are also formatted pretty poorly, https://github.com/qivicon/smarthome/blob/267d4d7b79c737ba4a81191a2ee465a7e94549d9/extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/constants.js

This is mostly a 2 tab indent spaced file. 

For example, if you copy the contents and paste it into http://jsbeautifier.org/ and beautify the code it changes the position of quite a few elements, you can also do this with HTML files but there isn't that big of a change when it comes to nesting elements because parsing angular JS code is difficult. It does resolve all of the issues of breaking elements in the middle. 

All that said, when it comes time to run the code I'm not sure if there is any downside to this formatting. Maybe adding a few extra spaces/lines to the HTML that will add .1 ms to the load time and the JS files are minified. I only raise concerns because of the readability when developing the files.

**Comment by kaikreuzer** [comment]
Are there maybe settings for the Eclipse JS formatter that should be improved then...?

**Comment by maggu2810** [comment]
@GradyD Could you have a look at the HTML settings (Web, HTML, ...) and JS (JavaScript, ...) in the Eclipse IDE preferences and improve them?

**Comment by aounhaider1** [comment]
Again, not my personal preferences. Just have been following what has been suggested by others. I applied all the available JS formatters (built-in + ESH) to the files to check if any of them is closer to that online "beautifier". None of them actually format the JS files like that. That said, if we really want our code to look like that we will have to customize either the ESH profile or create a new one.

**Comment by maggu2810** [comment]
> That said, if we really want our code to look like that we will have to customize either the ESH profile or create a new one.

It is no problem from my side to change the formatter settings.
There is not much (perhaps only some files) code that use it at the moment.

IMHO you (@aounhaider1, @GradyD) could play with the settings together to get a well defined one.

But we should use the Eclipse format feature to get a well defined layout of the sources.
For me it does not matter if it is equal to another one.
For me, all JS files should use the same format and all HTML should use the same format (and sure, all Java files should use a consistent format).

**Comment by GradyD** [comment]
Morning (afternoon?) guys,

I'm by no means a java/eclipse developer but I can dig around in the settings some point this week and see what I can come up with. I'm more then happy to work with @aounhaider1 on this if you have some experience with eclipse formatters? I have zero experience in eclipse but I have a good handle on what a formatted HTML/JS should look like and it appears that @aounhaider1 and I agree on what it should be formatted to.

**Comment by kaikreuzer** [comment]
Hi @GradyD, you do not need to be an Eclipse expert for this - simply go to Preferences->JavaScript->Formatter->Edit...:
screen shot 2016-02-11 at 08 47 20
and start playing around with the different options until you are satisfied with the result and tell us which settings should be changed.

**Comment by maggu2810** [comment]
Don't forget after 'Edit...' to export it to prevent loosing settings if the Eclipse IDE Oomph / Installer sync that settings after restart.

**Comment by GradyD** [comment]
Spent some time working on this effort this evening. I came up with the follow changes,

In Web > HTML Files > Editor I changed
* Changed line width to 999 (to prevent wrapping in middle of elements)
* Tabs to spaces
* Indentation size to 4 (4 spaces)
* Removed: a, img, input, label, li, select, span, td, and th from Inline Elements

For the Javascript I changed 
* Tab policy to spaces and indentation size to 4
* On Line Wrapping tab I changed Maximum line Width to 9999 to prevent awkward wrapping when using `function().function().function()` type of code

**Comment by kaikreuzer** [comment]
@GradyD: I have changed the settings according to your suggestions, see
https://github.com/eclipse/smarthome/pull/1084 and https://github.com/openhab/openhab-distro/pull/140.

Are we sure that we want spaces instead of tabs in HTML? I thought that we once said that these files go over the wire and thus tabs are more efficient.

**Comment by maggu2810** [comment]
> Are we sure that we want spaces instead of tabs in HTML? I thought that we once said that these files go over the wire and thus tabs are more efficient.

Yes, this have been decided some time ago.
If we minimize the JS on production builds, this does not care.
I don't care if that are spaces or tabs. I preferred tabs because the Paper UI used it in the most files.

**Comment by kaikreuzer** [comment]
Btw, @GradyD or @aounhaider1, Could any of you help @marcelrv on how to best get into Paper UI development by answering on https://www.eclipse.org/forums/index.php?t=rview&goto=1724075#msg_1724075? Maybe such info could even be added as a sub section on https://github.com/eclipse/smarthome/blob/master/docs/documentation/development/ide.md?

**Pull request closed without merging**